### ISSUE TITLE
Inform users about immer workings for complex objects

### ISSUE THREAD

**Issue by thibthib18** [ctrlplusb/easy-peasy#809]
Easy-peasy documentation is implying that mutations can be done on all parts of the state, without specifying that they are some [restrictions for Classes, Maps and Sets](https://immerjs.github.io/immer/complex-objects/)

I lost a few hours figuring out that my mutations on a class instance weren't triggering a new rendering because it was missing `[immerable]=true`. The documentation about [modifying the state](https://easy-peasy.vercel.app/docs/tutorials/primary-api.html#modifying-the-state) could be updated to warn other users about this. 

Something like:
"/!\ While this applies to array and plain JS objects, other objects may not be immutable by default. Refer to immer documentation for more info"

Many thanks for this library and your work.

**Comment by no-stack-dub-sack** [comment]
Thanks for the report @thibthib18, I'll try to create a code-sandbox to reproduce this soon (which I expect to be able to do from the immer docs and your experience) and then we can go from there on determining the best way to inform easy-peasy users about this.

**Comment by AnsonH** [comment]
I encountered the same issue a few months ago and I posted a self-answering Stack Overflow question:

[easy-peasy: useStoreState not working with ES6 class instances](https://stackoverflow.com/questions/74002866/easy-peasy-usestorestate-not-working-with-es6-class-instances)

I have included some Code Sandbox examples in the answer. Hope you'll find them useful during your investigation @no-stack-dub-sack :)

**Comment by no-stack-dub-sack** [comment]
Thanks @AnsonH! Slightly modified version of your sandbox here:

[![Edit Easy Peasy w/ ES6 classes (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/easy-peasy-w-es6-classes-forked-e1ixz5?fontsize=14&hidenavigation=1&theme=dark)

This sandbox illustrates the issue, and provides two possible solutions for getting this to work with classes.

The first, as the OP said, and as noted in your SO post, is to use the `immerable` symbol to mark the class as compatible with Immer:

```js
import { immerable } from "immer";

class Person {
  [immerable] = true;

  constructor(name, age) {
    this.name = name;
    this.age = age;
  }
}
```

The second, available to users of easy-peasy 5.2.0 and greater, is to disable and to write the immutable update yourself:

```js
  updatePersonName: action(
    (state, payload) => {
      return {
        ...state,
        person: new Person(payload, state.person.age),
      };
    },
    { immer: false }
  )
```

Thanks again for the report @thibthib18, I'll spin up a PR to call out this issue in the docs.

**Comment by jmyrland** [comment]
> Easy-peasy documentation is implying that mutations can be done on all parts of the state, without specifying that they are some [restrictions for Classes, Maps and Sets](https://immerjs.github.io/immer/complex-objects/)

As for `Map`s and `Set`s, you'd need import `easy-peasy/map-set-support`  as stated here: https://easy-peasy.dev/docs/introduction/browser-support.html#supporting-map-or-set-within-your-state

That being said, those docs are quite hidden - and we should probably add a reference to it [here](https://easy-peasy.vercel.app/docs/tutorials/primary-api.html#modifying-the-state).
